ethexe/ethereum/src/lib.rs

```diff
@@ -270,3 +271,40 @@ impl SignerSync for Sender {
         self.chain_id
     }
 }
+
+// TODO: Maybe better to append solution like this to alloy.
+trait TryGetReceipt<T: Transport + Clone, N: Network> {
+    /// Works like `self.get_receipt().await`, but retries a few times if rpc returns a null response.
+    async fn try_get_receipt(self) -> Result<N::ReceiptResponse>;
+}
+
+impl<T: Transport + Clone, N: Network> TryGetReceipt<T, N> for PendingTransactionBuilder<'_, T, N> {
+    async fn try_get_receipt(self) -> Result<N::ReceiptResponse> {
+        let tx_hash = *self.tx_hash();
+        let provider = self.provider().clone();
+
+        let mut err = match self.get_receipt().await {
```

Comment:
Wdyt to put initial query in loop so simplify code

Reply:
It's impossible because initial query is different and borrows the self